### PR TITLE
Ak adding cart api

### PR DESCRIPTION
**Pull request by aqattan**

### CODE REVIEW

File: app/controllers/api/v1/carts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        cart_items = []
+        cart.line_items.each do |item|
+          cart_items << CartSerializer.new(item, action_name: 'show')
+        end
+        hash_to_return = {items: cart_items, kidu_balance: kidu_balance,
+                          order_total: order_total, new_balance: new_balance,
+                          place_order: place_order, back_button: '/products'}
+        render json: hash_to_return, status: 200, action_name: 'show'
+      else
+        hash_to_return = {kidu_balance: kidu_balance, message: I18n.translate('views.cart.show.message.your_cart_empty'), back_button: '/products'}
```
Comment (ssharba) [inline app/controllers/api/v1/carts_controller.rb]: @aqattan what is the back_button for api?

Comment (aqattan) [reply app/controllers/api/v1/carts_controller.rb]: @ssharba I added it as a reference  for the mobile team if they need a back link

Comment (ssharba) [reply app/controllers/api/v1/carts_controller.rb]: unless they ask for, don't add any extra values

File: app/controllers/api/v1/carts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class Api::V1::CartsController < ApplicationController
+  def show
+    begin
+      cart = Cart.find_by(id: params[:id], customer_id: params[:customer_id])
+      kidu_balance = CustomerInfo.find_by(customer_id: params[:customer_id]).try(:token_balance)
+      if cart.line_items.present?
+        order_total = Helper.display_token_value(cart.line_items.map(&:total_price).reduce(:+))
+        new_balance = (kidu_balance - order_total).round(2)
+        if order_total > kidu_balance
```
Comment (ssharba) [inline app/controllers/api/v1/carts_controller.rb]: @aqattan show action will only return info no likes for next actions

File: app/serializers/cart_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class CartSerializer < ActiveModel::Serializer
+  attributes :product, :quantity, :show_product_link, :delete_item_from_cart
+
+  def product
+    {id: object.product.id, name: object.product.name, description: object.product.description,
+     price: price_after_convert, total_quantity: object.product.quantity,
+    first_image: object.product.images.first.attachment}
+  end
+  def show_product_link
```
Comment (ssharba) [inline app/serializers/cart_serializer.rb]: remove this too please

File: app/controllers/api/v1/carts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+class Api::V1::CartsController < ApplicationController
+  def show
+    begin
+      cart = Cart.find_by(id: params[:id], customer_id: params[:customer_id])
+      kidu_balance = CustomerInfo.find_by(customer_id: params[:customer_id]).try(:token_balance)
+      if cart.line_items.present?
```
Comment (ssharba) [inline app/controllers/api/v1/carts_controller.rb]: line item has been changed to order item, let's hold this PR until we merge tracking branch in the UI

Comment (aqattan) [reply app/controllers/api/v1/carts_controller.rb]: Okay

Comment (ssharba) [reply app/controllers/api/v1/carts_controller.rb]: go a head and test it now

**Pull request merged**